include/seqan3/range/container/small_string.hpp

```diff
@@ -158,11 +158,12 @@ class small_string : public small_vector<char, capacity_ + 1>
      *
      * No-throw guarantee if value_type is std::is_nothrow_copy_constructible.
      */
-    template <std::forward_iterator begin_it_type, std::sentinel_for<begin_it_type> end_it_type>
-    constexpr void assign(begin_it_type begin_it, end_it_type end_it) noexcept
+    template <std::forward_iterator begin_it_type, typename end_it_type>
     //!\cond
-        requires std::constructible_from<value_type, /*ranges::iter_reference_t*/reference_t<begin_it_type>>
+        requires std::sentinel_for<end_it_type, begin_it_type> &&
+                 std::constructible_from<value_type, /*ranges::iter_reference_t*/reference_t<begin_it_type>>
```

Comment:
```suggestion
                 std::constructible_from<value_type, std::ranges::iter_reference_t<begin_it_type>>
```
I think this should work now?

Reply:
thank you for the catch, I look into it.

Reply:
I just saw that I didn't comment this out, so this is not in the scope of my PR.